File: src/main/java/com/imcode/imcms/maintenance/MaintenanceFilter.java
package com.imcode.imcms.maintenance;

import javax.servlet.*;
import java.io.IOException;

/**
 * 
 */
public class MaintenanceFilter implements Filter {

    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
        servletRequest.getRequestDispatcher("/WEB-INF/maintenance/index.jsp").forward(servletRequest, servletResponse);
    }

    public void destroy() {
        
    }

    public void init(FilterConfig filterConfig) throws ServletException {

    }    
}

File: src/main/clojure/com/imcode/imcms/maintenance/MaintenanceFilter.clj
(ns com.imcode.imcms.maintenance.MaintenanceFilter
  #^{:doc "Maintenance filter. See mapping configuration in WEB-INF/web.xml."}
  (:gen-class
    :implements [javax.servlet.Filter]
    :init init-instance)

  (:require
    [com.imcode.imcms.backdoor.controller :as controller]))

(defn -init-instance []
  [[] nil])

(defn -init [this filterConfig])
(defn -destroy [this])

(defn -doFilter [this request response filterChain]
  (controller/handle request response))